SpatialSameResponseNormalization.lua

```diff
@@ -11,34 +11,17 @@ function SpatialSameResponseNormalization:__init(size, alpha, beta)
   
   local numerator = nn.Identity()
   local denominator = nn.Sequential()
-  denominator:add(nn.SpatialZeroPadding(pad, pad, pad, pad))
-  denominator:add(nn.Power(2))
-  denominator:add(nn.SpatialAveragePooling(self.size,self.size,1,1))
-  denominator:add(nn.MulConstant(self.alpha,true))
-  denominator:add(nn.AddConstant(1,true))
-  denominator:add(nn.Power(self.beta))
+    :add(nn.Power(2))
+    :add(nn.SpatialAveragePooling(self.size,self.size,1,1,pad,pad))
+    :add(nn.MulConstant(self.alpha,true))
+    :add(nn.AddConstant(1,true))
+    :add(nn.Power(self.beta))
 
   local divide = nn.ConcatTable()
   divide:add(numerator)
   divide:add(denominator)
   
-  self.modules = nn.Sequential()
-  self.modules:add(divide)
-  self.modules:add(nn.CDivTable())
+  table.insert(self.modules, divide)
+  table.insert(self.modules, nn.CDivTable())
```

Comment:
Why don't you use the `:add` method ? Just inserting in the table the modules won't update the `self.output` and `self.gradInput` variables, right ?